Support non-Android Things devices too

daj opened the pull request:
Because normal phones are important too :-)

TESTING

1) Nexus 5 emulator (Android 7.0, API 24).  You can touch anywhere on the screen to take the photo:

<img width="434" alt="screen shot 2017-11-28 at 10 22 54 pm" src="https://user-images.githubusercontent.com/739125/33356517-e6fc1d20-d48a-11e7-987b-d298bd8d6874.png">

<img width="435" alt="screen shot 2017-11-28 at 10 23 07 pm" src="https://user-images.githubusercontent.com/739125/33356523-f4d564ce-d48a-11e7-8fb3-d08151d7830b.png">

2) Google Iot_imx7d_pico (Android 7.0, API 24).  The hardware button works fine, and the screen shows the results.

![img-2103](https://user-images.githubusercontent.com/739125/33052723-2f4146e4-ce3e-11e7-8cd0-92e34466826e.JPG)

jdkoren commented on app/src/main/java/com/example/androidthings/imageclassifier/ImageClassifierActivity.java:
```diff
@@ -80,6 +83,20 @@ protected void onCreate(final Bundle savedInstanceState) {
         mResultViews[0] = (TextView) findViewById(R.id.result1);
         mResultViews[1] = (TextView) findViewById(R.id.result2);
         mResultViews[2] = (TextView) findViewById(R.id.result3);
+        mButton = (android.widget.Button) findViewById(R.id.takePhoto);
```
The button isn't necessary here, you can just use `findViewById(...).setOnClickListener(...)`

jdkoren commented on app/src/main/res/layout/activity_camera.xml:
```diff
@@ -58,4 +58,13 @@
+    <Button
+        android:id="@+id/takePhoto"
+        android:text="Take Photo"
+        android:layout_width="160dip"
```
Prefer `wrap_content` over hardcoded values. You can always use padding to artificially make the view larger, but I don't think that's needed.

jdkoren commented on app/src/main/res/layout/activity_camera.xml:
```diff
@@ -58,4 +58,13 @@
+        android:id="@+id/takePhoto"
+        android:text="Take Photo"
+        android:layout_width="160dip"
+        android:layout_height="40dip"
```
`wrap_content`

jdkoren commented on app/src/main/java/com/example/androidthings/imageclassifier/ImageClassifierActivity.java:
```diff
@@ -313,4 +335,16 @@ private void requestPermission() {
+     */
+    private boolean isAndroidThingsDevice(Context context) {
+        final PackageManager pm = context.getPackageManager();
+        boolean isRunningAndroidThings = pm.hasSystemFeature("android.hardware.type.embedded");
```
change to [`PackageManager.FEATURE_EMBEDDED`](https://developer.android.com/reference/android/content/pm/PackageManager.html#FEATURE_EMBEDDED)

daj replied:
FEATURE_EMBEDDED was added in API level 26.  This project currently has minSdkVersion 24, which means that FEATURE_EMBEDDED is not available.  Do you want the minSdkVersion and targetSdkVersion updated to 26, or are you happy to keep the literal?  For now I'll keep the literal value, and add a comment explaining it.

jdkoren replied:
It actually depends on the `compileSdkVersion` of the project. As long as that is at least 26, you could reference the constant and it will get inlined at compile time (so it won't be an issue at runtime even on a lower platform level). Since this project has `compileSdkVersion 25`, let's leave it alone for now.

jdkoren reviewed (changes requested):
Sorry for the delay. This is a very useful change, just a few small things and we'll be really happy to have it :)

daj committed:
Support non-Android Things devices too

Because normal phones are important too :-)

Tested on emulator: Nexus 5 (Android 7.0, API 24)

daj commented:
I've updated the pull request description and code and reran the manual tests.  I added some extra logic to allow users to take a photo even without a hardware button.  I think we should be good to merge now.

FYI, I used this sample in a blog post here: https://medium.com/@daj/tensorflow-image-classifiers-on-android-android-things-and-ios-82cee88095d3

mangini commented on app/src/main/java/com/example/androidthings/imageclassifier/ImageClassifierActivity.java:
```diff
@@ -72,19 +75,44 @@ protected void onCreate(final Bundle savedInstanceState) {
+        View takePhotoButton = findViewById(R.id.takePhoto);
+
+        if (!isAndroidThingsDevice(this)) {
+            // Android Things devices have an LED and a hardware button.  On other devices (like
```
Android Things devices don't necessarily have an LED and a hardware button, and this onscreen button can be useful if a user is running the app on an AT board without special buttons/LEDs. I would always show it, even if it's AT.

daj replied:
My display isn't a touchscreen, so I found it confusing to show a button that you can't touch to press.  Are you sure you want it displayed even when you can't press it?  Is there a better test I can do here?

mangini replied:
You can still connect a pointing device, like a USB mouse, but I see your point. Perhaps you can remove the button and take a photo when the user taps on the photo area, so that the UI is not confusing to users without touching screens or pointing devices, but still work well for users with a touchscreen.

daj replied:
Good idea, thanks for the suggestion!  Ok, I've made the update and squashed the commits.  I retested on an emulator and I can touch anywhere to take the photo.

mangini commented on app/src/main/res/values/strings.xml:
```diff
@@ -19,4 +19,5 @@
     <string name="request_permission">This sample needs camera permission.</string>
     <string name="camera_error">This device doesn\'t support Camera2 API.</string>
     <string name="help_message">When the LED is on, press the button to capture an image</string>
+    <string name="take_photo">Take Photo</string>
```
I think this is not needed anymore, right?

daj replied:
Agreed, thanks!

mangini commented on app/src/main/res/layout/activity_camera.xml:
```diff
@@ -58,4 +58,5 @@
             android:textColor="@android:color/black"
             tools:text="cat: 0.21423423" />
     </LinearLayout>
+
```
empty line

daj replied:
Removed...even though I prefer the newline here. :-)

mangini commented on app/src/main/java/com/example/androidthings/imageclassifier/ImageClassifierActivity.java:
```diff
@@ -73,18 +76,37 @@ protected void onCreate(final Bundle savedInstanceState) {
         mResultViews[1] = (TextView) findViewById(R.id.result2);
         mResultViews[2] = (TextView) findViewById(R.id.result3);
 
+        // Let the user touch the screen to take a photo
```
To avoid a weird race condition where the background handler is not initialized for some reason but this click listener is still set, I recommend you to move this setOnClickListener to the end of the init() method.

daj replied:
Done.

mangini reviewed (changes requested):
Almost there, just a few minor comments. Thanks!

mangini reviewed (approved):
Thanks for making the changes!
